kernel/src/sched.rs

```diff
@@ -70,17 +70,37 @@ impl Kernel {
     }
 
     /// Run a closure on a specific process if it exists. If the process does
-    /// not exist (i.e. it is `None` in the `processes` array) then `default`
-    /// will be returned. Otherwise the closure will executed and passed a
-    /// reference to the process.
-    crate fn process_map_or<F, R>(&self, default: R, process_index: usize, closure: F) -> R
+    /// not exist (i.e. it is `None` in the `processes` array, or has stopped,
```

Comment:
Is it true that if a process has been stopped this will return `default`? From looking through it seems that a stopped process still has a valid and unchanged AppId, and this function does not seem to call anything that checks if a process has been stopped.

Reply:
Ah ok good catch! This raises a case I hadn't really considered...what should happen if an app is in the `StoppedFaulted` state (or perhaps an `Ended` state if one were to exist)? The process is still there, and its the same process so its identifier probably shouldn't change, and certain capsules may want to be able to see that it exists (like process console). However, other capsules should be able to know it is no longer active (and will never be again). Perhaps `process_map_or()` should not change, but users of `process_map_or()` may need to check that the process is still active. In particular grant.enter() shouldn't succeed on an ended process.

In any case I will update the comment.

Reply:
I updated the comment.